Implementation of thecorresponding item in docs/api-wish-list.txt.
Should be safe for application in development and stable branch (minor
release candidate).

  Lars J

Index: include/Inventor/SoType.h
===================================================================
RCS file: /export/cvsroot/Coin/include/Inventor/SoType.h,v
retrieving revision 1.19
diff -u -r1.19 SoType.h
--- include/Inventor/SoType.h	23 Nov 2001 14:25:46 -0000	1.19
+++ include/Inventor/SoType.h	22 Jan 2002 15:20:03 -0000
@@ -77,6 +77,8 @@
 
   static int getNumTypes(void);
 
+  instantiationMethod getInstantiationMethod(void) const;
+
 private:
   static void clean(void);
 
Index: src/misc/SoType.cpp
===================================================================
RCS file: /export/cvsroot/Coin/src/misc/SoType.cpp,v
retrieving revision 1.23
diff -u -r1.23 SoType.cpp
--- src/misc/SoType.cpp	20 Jan 2002 12:35:01 -0000	1.23
+++ src/misc/SoType.cpp	22 Jan 2002 15:20:03 -0000
@@ -420,6 +420,19 @@
 }
 
 /*!
+  Returns a pointer to the method used to instantiate objects of the given
+  type.
+
+  \since 2002-01-22
+*/
+
+SoType::instantiationMethod
+SoType::getInstantiationMethod(void) const
+{
+  return (*SoType::typedatalist)[(int)this->getKey()]->method;
+}
+
+/*!
   \fn int16_t SoType::getKey(void) const
 
   This method returns the type's index in the internal typelist.
